tests/test_pumptrap_calibration.py

```diff
@@ -0,0 +1,274 @@
+import os
+import glob
+import numpy as np
+from corgidrp.mocks import generate_mock_pump_trap_data
+from corgidrp.detector import imaging_area_geometry
+from corgidrp.data import Dataset
+from corgidrp.l1_to_l2a import prescan_biassub
+from corgidrp.l2a_to_l2b import em_gain_division
+from corgidrp.pump_trap_calibration import tpump_analysis, tau_temp
+
+
+
+# Adjust the system's limit of open files. We need to load 2000 files at once. 
+# some systems don't like that. 
+import resource
+soft_limit, hard_limit = resource.getrlimit(resource.RLIMIT_NOFILE)
+resource.setrlimit(resource.RLIMIT_NOFILE, (6001, hard_limit))
+
+def rebuild_dict(trap_pump_array):
+        '''
+        Partially rebuild the trap_dictionary from the trap_pump_array to help with testing
+
+        Args:
+            trap_pump_array: array of trap_pump objects
+        
+        Returns:
+            trap_dict: dictionary of trap_pump objects
+
+        '''
+        trap_dict = {}
+
+        electrode_dict = {"LHS": 10, 
+                        "CEN": 20,
+                        "RHS": 30}
+        
+        electrode_dict_inverse = {10: "LHS",
+                                20: "CEN",
+                                30: "RHS"}
+        
+        
+        for trap_pump in trap_pump_array:
+            electrode_key = int(((trap_pump[2] // 10) % 10)*10)
+            electrode_number = int(trap_pump[2] % 10)
+            electrode_string = electrode_dict_inverse[electrode_key]+"el"+str(electrode_number)
+            key = ((trap_pump[0],trap_pump[1]), electrode_string, int(trap_pump[3]))
+            
+            trap_dict[key] = {}
+            trap_dict[key]['cap']  = [trap_pump[4], 0, trap_pump[5]] #Add the error in to keep the expected dimensions
+            trap_dict[key]['E'] = trap_pump[6]
+            trap_dict[key]['cs'] = trap_pump[7]
+            trap_dict[key]['Rsq'] = trap_pump[8]
+            trap_dict[key]['tau at input T'] = trap_pump[9]
+
+        return trap_dict
+
+def test_tpump_analysis():
+    '''
+    After reading in the data and some setup, then run the subset of appropriate tests from
+    test_tfit_const_True_sub_noise_ill in ut_tpump_final.py
+    '''
+
+    # Set the seed - II&T ut tests don't work everytime, so let's fix it. 
+    np.random.seed(39)
+
+    #Generate the mock data:
+    test_data_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'test_data', "pump_trap_data_test")
+    metadata_file = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'test_data', "metadata_test.yaml")
+    print("Generating mock data")
+    generate_mock_pump_trap_data(test_data_dir, metadata_file)
+    print("Done generating mock data")
```

Comment:
Not sure if we're okay with prints in pytest outputs?  If we are, that's fine.

Reply:
It's fine. They're neither warnings or errors, and will only be displayed if the test fails.